### PR TITLE
Add BroadcastGap domain object

### PR DESCRIPTION
**Pull request by BPScott**
This is object similar to Broadcast, that can be used for handling
gaps in service transmission schedules

### CODE REVIEW

File: tests/Domain/Entity/BroadcastGapTest.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+use BBC\ProgrammesPagesService\Domain\Entity\BroadcastGap;
+use DateTimeImmutable;
+use PHPUnit\Framework\TestCase;
+
+class BroadcastGapTest extends TestCase
+{
+    public function testConstructorRequiredArgs()
+    {
+        $service = $this->createMock('BBC\ProgrammesPagesService\Domain\Entity\Service');
```
Comment (votemike) [inline tests/Domain/Entity/BroadcastGapTest.php]: `$service = $this->createMock(Service::class);`?

File: tests/Domain/Entity/BroadcastGapTest.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        $startAt = new DateTimeImmutable('2015-01-01 06:00:00');
```
Comment (votemike) [inline tests/Domain/Entity/BroadcastGapTest.php]: Should these have `Z` on the end?

Comment (BPScott) [inline tests/Domain/Entity/BroadcastGapTest.php]: Doesn't really matter, nothing else in the Domain tests bother

File: tests/Domain/Entity/BroadcastGapTest.php (hunk 1 above)
Comment (BPScott) [inline tests/Domain/Entity/BroadcastGapTest.php]: None of the other Domain tests use this still, but they probably should.

**Pull request merged**